src/components/Geogebra/GeogebraViews.js

```diff
@@ -74,15 +62,19 @@ export default class {
             }
         });
 
+        api.service('elements').on('removed', (element) => {
+            const pos = this.workshopIds.indexOf(element.workshop);
```

Comment:
Add one line before new block. Change `pos` to `index`.

Reply:
Sth like this?

```
api.service('elements').on('removed', (element) => {
            const index = this.workshopIds.indexOf(element.workshop);

            if (index !== -1) {
                this.GAVs[index].deleteObject(element.name);
            }
        });
```

Reply:
Yes.